app/session.server.ts

```diff
@@ -59,6 +60,12 @@ export async function requireUsername(
     const searchParams = new URLSearchParams([['redirectTo', redirectTo]]);
     throw redirect(`/login?${searchParams}`);
   }
+
+  // Logout user if they are deactivated
```

Comment:
I wonder if you should do this in `getUsername` instead, so it happens earlier, and for all paths where we get a user/username?

Reply:
I went with `requireUsername` because `getUsername` seemed like only getting the username instead of requiring a valid username. But I will look into the paths to see if it is better to move this logic there